Sources/SwiftSyntax/Syntax.swift

```diff
@@ -380,6 +384,57 @@ extension Syntax {
   }
 }
 
+/// Temporary non-owning Syntax.
+///
+/// This can be used for handing Syntax node without ARC traffic.
+struct UnownedSyntax {
+  var raw: RawSyntax
+  var info: Unmanaged<Syntax.Info>
+
+  @_transparent
+  init(_ node: __shared Syntax) {
+    self.raw = node.raw
+    self.info = .passUnretained(node.info.unsafelyUnwrapped)
+  }
+
+  /// Extract the Syntax value.
+  @_transparent
+  var value: Syntax {
+    Syntax(raw, info: info.takeUnretainedValue())
+  }
+
+  /// Get the parent of the Syntax value, but without retaining it.
+  @_transparent
+  var parent: UnownedSyntax? {
+    return info._withUnsafeGuaranteedRef {
+      switch $0.info.unsafelyUnwrapped {
+      case .nonRoot(let info):
+        return UnownedSyntax(info.parent)
+      case .root(_):
```

Comment:
```suggestion
      case .root:
```

Reply:
Looks like we're inconsistent. `var nonRootInfo` uses `case .root(_)`